internal/pkg/build/apps/apps.go

```diff
@@ -47,6 +48,8 @@ var (
 	}
 )
 
+var reg = regexp.MustCompile(`[^a-zA-Z0-9_]`)
```

Comment:
In the SCI-F spec it states:

https://sci-f.github.io/spec-v1
> illegal characters or uppercase in app names is not allowed

should this change to remove the A-Z? 

It doesn't define `illegal characters`.

Reply:
It's the whole point, when building with 2.6 and setting a app name `test.app` or `test-app`, the resulting variables always end with `SCIF_APPROOT_test_app` ... Always with 2.6 you can even set app name to `test,app` and screw up environment variables, you can also define app name with uppercase ... so looks like there is not much validation but some "illegal" characters are replaced by `_` when generating env vars `SCIF_APPXXXX_app_name`.

Reply:
Understood. The spec is not very specific, so just adopting the 2.6 style stuff sounds good to me.